src/sys/socket/addr.rs

```diff
@@ -15,12 +17,24 @@ use std::os::unix::ffi::OsStrExt;
  *
  */
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+#[repr(C)]
+pub struct sockaddr_nl {
```

Comment:
libc already defines `Clone` and `Copy` for most or all of its structs (see s! macro: http://rust-lang-nursery.github.io/libc/x86_64-unknown-linux-gnu/src/libc/macros.rs.html#42). So unless `PartialEq`, `Eq`, `Debug` and `Hash` are necessary, it should be fine to use the libc version.